database_size/report/ir_model_size_report.py

```diff
@@ -0,0 +1,232 @@
+                ),
+            ),
+        ]
+        action = self.env["ir.actions.actions"]._for_xml_id(
+            "database_size.ir_model_size_action"
+        )
+        action["domain"] = domain
+        return action
+
+    @api.model
+    def _move_dates_to_context(self, domain):
+        """Move the requested comparison date from the domain into the context.
+
+        The values in the context will be used when creating the virtual table
+        in `_table_query`.
+        """
+        new_domain = []
+        values = {}
+        for clause in domain or []:
+            for field in ("measurement_date", "historical_measurement_date"):
+                if not isinstance(clause, tuple | list) or clause[0] != field:
+                    continue
+                if field in values:
+                    raise UserError(
+                        self.env._(
+                            f"You cannot search on more than one value for {field} "
+                            "at the same time."
+                        )
+                    )
+                if clause[1] in ("=", "==") and clause[2]:
+                    values[field] = clause[2]
+                else:
+                    raise UserError(
+                        self.env._(
+                            f"Searching {field} for '{clause[1]} {clause[2]}' is "
+                            "not supported."
+                        )
+                    )
+                new_domain.append((1, "=", 1))
+            else:
+                new_domain.append(clause)
+        if values:
+            self = self.with_context(**values)
+        return self, new_domain
+
+    @api.model
+    def _where_calc(self, domain, active_test=True):
+        """Move the requested dates from the domain into the context"""
+        (self, new_domain) = self._move_dates_to_context(domain)
+        return super()._where_calc(new_domain, active_test=active_test)
+
+    @api.model
+    def search(self, domain, offset=0, limit=None, order=None):
+        """Move the requested dates from the domain into the context"""
+        (self, new_domain) = self._move_dates_to_context(domain)
+        return super().search(new_domain, offset=offset, limit=limit, order=order)
+
+    @api.model
+    def search_count(self, domain, limit=None):
+        """Move the requested dates from the domain into the context"""
+        (self, new_domain) = self._move_dates_to_context(domain)
+        return super().search_count(new_domain, limit=limit)
+
+    @property
+    def _table_query(self):
+        """Report comparative database size changes between two dates.
+
+        The dates are inserted in the context in this model's `search` method.
+        """
+        measurement_date = self.env.context.get("measurement_date")
+        if measurement_date:
+            if isinstance(measurement_date, str):
+                measurement_date = fields.Date.to_date(measurement_date)
```

Comment:
I'd call to_date unconditionally, it does the right thing when you pass a date/datetime there. also below

Reply:
Wasn't aware, thanks!